Lib/SharedSet.hpp

```diff
@@ -525,7 +525,7 @@ class SharedSet {
   }
 
   auto iter() const
-  { return arrayIter(_items, size()); }
+  { return arrayIter(static_cast<const T *>(_items), size()); }
```

Comment:
This is declared as `T _items[1];`, but is used as a dynamic-length array with `operator new`. The generated iterator then captures the type `T[1] &`, and of course indexes greater than `0`, which is what UBSan catches. I'm not sure about what the standard says about this, but it's a bit dodgy so let's not. Instead cast to pointer (morally `T[]`) and do what was expected.